src/bin/julialauncher.rs

```diff
@@ -166,32 +166,49 @@ fn get_julia_path_from_channel(
     juliaupconfig_path: &Path,
     juliaup_channel_source: JuliaupChannelSource,
 ) -> Result<(PathBuf, Vec<String>)> {
+    let potential_matches: Vec<_> = config_data
+        .installed_channels
+        .keys()
+        .filter(|&item| {
+            item.starts_with(channel) && !item.starts_with("0") && !item.starts_with("1")
+        })
+        .cloned()
+        .collect();
+    let actual_channel = if potential_matches.len() == 1 {
+        potential_matches
+            .first()
+            .expect("length should be 1")
+            .as_str()
+    } else {
+        channel
+    };
```

Comment:
A more efficient solution (though probably negligibly so)

```suggestion
    let actual_channel = if channel
        .chars()
        .next()
        .map(char::is_numeric)
        .unwrap_or(false)
    {
        channel
    } else {
        match only(
            config_data
                .installed_channels
                .keys()
                .filter(|&item| item.starts_with(channel)),
        ) {
            Some(val) => val,
            None => channel,
        }
    };
```

With a helper function

```rs
fn only<T, I>(mut iter: I) -> Option<T>
where
    I: Iterator<Item = T>,
{
    let first = iter.next();
    if iter.next().is_some() {
        // There are multiple elements
        return None;
    }
    first
}
```

Mostly this is a stylistic choice.